query-service-impl/src/test/java/org/hypertrace/core/query/service/postgres/QueryRequestToPostgresSQLConverterTest.java

```diff
@@ -1150,13 +1151,143 @@ void testQueryWithAverageRateInOrderBy() {
+                createAliasedFunctionExpression(
+                    "MAX", "Span.duration_millis", "avg_duration_millis"))
+            .setFilter(
+                Filter.newBuilder()
+                    .setOperator(Operator.AND)
+                    .addChildFilter(startTimeFilter)
+                    .addChildFilter(endTimeFilter)
+                    .build())
+            .addOrderBy(
+                createOrderByExpression(
+                    createAliasedFunctionExpression(
+                        "DISTINCTCOUNT", "Span.labels", "distinctcount_labels"),
+                    SortOrder.ASC))
+            .setLimit(15)
+            .build();
+
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        queryRequest,
+        "select encode(span_id, 'hex'), count(distinct column2),"
+            + " avg(duration_millis), max(duration_millis) FROM"
+            + " ( select span_id, unnest(labels) as column2, duration_millis from public.\"span-event-view\""
+            + " where "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "and ( start_time_millis > 1570658506605 and end_time_millis < 1570744906673 )"
+            + " ) as intermediate_table"
+            + " group by span_id order by 2 limit 15",
+        tableDefinition,
+        executionContext);
+  }
+
+  @Test
+  void testQueryWithPercentileForTdigestColumn() {
+    Filter startTimeFilter =
+        createTimeFilter("Span.start_time_millis", Operator.GT, 1570658506605L);
+    Filter endTimeFilter = createTimeFilter("Span.end_time_millis", Operator.LT, 1570744906673L);
+    QueryRequest queryRequest =
+        QueryRequest.newBuilder()
+            .addSelection(createColumnExpression("Span.id"))
+            .addGroupBy(createColumnExpression("Span.id"))
+            .addAggregation(
+                createAliasedFunctionExpression(
+                    "PERCENTILE99", "Span.response_time_millis", "P99_response_time"))
+            .setFilter(
+                Filter.newBuilder()
+                    .setOperator(Operator.AND)
+                    .addChildFilter(startTimeFilter)
+                    .addChildFilter(endTimeFilter)
+                    .build())
+            .setLimit(15)
+            .build();
+
+    TableDefinition tableDefinition = getDefaultTableDefinition();
+    defaultMockingForExecutionContext();
+
+    assertSQLQuery(
+        queryRequest,
+        "select encode(span_id, 'hex'), tdigest_percentile(response_time_millis_tdigest,0.99)"
+            + " from public.\"span-event-view\""
+            + " where "
+            + tableDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "and ( start_time_millis > 1570658506605 and end_time_millis < 1570744906673 )"
+            + " group by span_id limit 15",
```

Comment:
Here, instead of `1`, it's a name pattern - are we missing something?

Reply:
Only columns in select query are replaced with numbers